			   READLINE PATCH REPORT
			   =====================

Readline-Release: 8.2
Patch-ID: readline82-002

Bug-Reported-by:	srobertson@peratonlabs.com
Bug-Reference-ID:
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-bash/2022-09/msg00049.html

Bug-Description:

It's possible for readline to try to zero out a line that's not null-
terminated, leading to a memory fault.

Patch (apply with `patch -p0'):

Upstream-Status: Submitted [https://ftp.gnu.org/gnu/readline/readline-8.2-patches/]
Signed-off-by: Khem Raj <raj.khem@gmail.com>

*** ../readline-8.2-patched/display.c	2022-04-05 10:47:31.000000000 -0400
--- display.c	2022-12-13 13:11:22.000000000 -0500
***************
*** 2684,2692 ****
  
    if (visible_line)
!     {
!       temp = visible_line;
!       while (*temp)
! 	*temp++ = '\0';
!     }
    rl_on_new_line ();
    forced_display++;
--- 2735,2740 ----
  
    if (visible_line)
!     memset (visible_line, 0, line_size);
! 
    rl_on_new_line ();
    forced_display++;
*** ../readline-8.2/patchlevel	2013-11-15 08:11:11.000000000 -0500
--- patchlevel	2014-03-21 08:28:40.000000000 -0400
***************
*** 1,3 ****
  # Do not edit -- exists only for use by patch
  
! 1
--- 1,3 ----
  # Do not edit -- exists only for use by patch
  
! 2
